### PR TITLE
[Draft] Set parameters result example

### PR DESCRIPTION
**Pull request by doisyg**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-planning/navigation2/issues/956 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Wyca's Elodie|

---

## Description of contribution in a few bullet points

As discussed here https://github.com/ros-planning/navigation2/issues/956

### CODE REVIEW

File: nav2_smac_planner/src/smac_planner_2d.cpp [hunk 1]
```diff
@@ -340,45 +334,46 @@ nav_msgs::msg::Path SmacPlanner2D::createPlan(
   return plan;
 }
 
-void SmacPlanner2D::on_parameter_event_callback(
-  const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
+rcl_interfaces::msg::SetParametersResult
+SmacPlanner2D::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameters)
 {
+  auto result = rcl_interfaces::msg::SetParametersResult();
```
Comment (SteveMacenski) [inline nav2_smac_planner/src/smac_planner_2d.cpp:340]: err, umm maybe `rcl_interfaces::msg::SetParametersResult result` instead? I think you've been writing some python code recently :wink:

File: nav2_smac_planner/src/smac_planner_2d.cpp [hunk 2]
```diff
@@ -340,45 +334,46 @@ nav_msgs::msg::Path SmacPlanner2D::createPlan(
...
+  RCLCPP_INFO_STREAM(_logger, "Param callback, size: " << parameters.size());
```
Comment (SteveMacenski) [inline nav2_smac_planner/src/smac_planner_2d.cpp:341]: Please replace with `RCLCPP_INFO` call instead using `%i`. Its a stylistic thing, I like to keep logging statements consistent within a project such that we can easily scan through code and not have to think too much. This is really simple, but that's how the creep starts :-)

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM, just small changes. Should we be doing anything to check if the event is a declaration (and to block an update in that case?)

File: nav2_smac_planner/src/smac_planner_2d.cpp (hunk 1 above)
Comment (doisyg) [reply nav2_smac_planner/src/smac_planner_2d.cpp:340]: hum, at least from cpp 11 using `auto` is pretty convenient, and I was just reproducing rclcpp style:
https://github.com/ros2/rclcpp/blob/d5ec2580805c3abd8b6fecaf3d09040e00e9f15e/rclcpp/src/rclcpp/parameter_service.cpp#L86
But no problem to use `rcl_interfaces::msg::SetParametersResult result` instead if you want

File: nav2_smac_planner/src/smac_planner_2d.cpp (hunk 2 above)
Comment (doisyg) [reply nav2_smac_planner/src/smac_planner_2d.cpp]: Debug artefact, removing it

**Comment by doisyg** [comment]
> Should we be doing anything to check if the event is a declaration (and to block an update in that case?)

I believe registering the callback after all the declaration is enough, no ? Plus I don't see how to check if it is a declaration in this callback

File: nav2_smac_planner/src/smac_planner_2d.cpp (hunk 1 above)
Comment (SteveMacenski) [reply nav2_smac_planner/src/smac_planner_2d.cpp:340]: Do `rcl_interfaces::msg::SetParametersResult result;`, its more compact. Clearly whoever wrote that was thinking in python. Its the same thing without `auto` or `()`, making it more compact. There's no reason to use auto for the default constructor.

Comment (doisyg) [reply nav2_smac_planner/src/smac_planner_2d.cpp:340]: :man_facepalming: nice catch, did not realize it. Will change at next push

**Comment by doisyg** [comment]
> Into the on activate state transition? That way its definitely after parameter declarations (and this is via a network interface, it shouldn't be allowed in the inactive states). Probably belongs there anyway.

I did not think of this before, but indeed, that looks like the perfect place to put parameters callback registration. I guess the handler should be reset in the deactivate transition too. Changes pushed

**Comment by SteveMacenski** [comment]
I didn't really think about it either until we started this discussion. Yeah, it should be reset in deactivate

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**